modules/tile-converter/src/i3s-converter/i3s-converter.ts

```diff
@@ -1155,39 +1155,41 @@ export default class I3SConverter {
    * @param propertyTable - Table with layer meta data.
    */
   private _convertPropertyTableToNodeAttributes(propertyTable: FeatureTableJson): void {
+    /*
+      We will append attributes if only the attributeStorageInfo doesn't contain anything.
+      If it already has attributes, return from the function.
+      According to ver 1.9 (see https://github.com/Esri/i3s-spec/blob/master/docs/1.9/attributeStorageInfo.cmn.md):
+      "The attributeStorageInfo object describes the structure of the binary attribute data resource of a layer,
+      which is the same for every node in the layer."
+    */
+    if (this.layers0!.attributeStorageInfo?.length) {
```

Comment:
It is the previous version of logic here. The `attributeStorageInfo` is created from first file which has `propertyTable`. That is not correct if we have extensions. I thought about different logic:
* If there is an `EXT_feature_metadata` or `EXT_structural_meatadata` extension, `attributeStorageInfo` is built from the extension JSON metadata. We have there all properties names and datatypes. `attributeStorageInfo` is not updated from `propertyTable` during conversion;
* If there is no any metadata extension, use existing logic, create `attributeStorageInfo` from `propertyTable`, accumulate it from all files, not only from the first one.

Reply:
Done